packages/nextjs-mf/utils/createDynamicFederatedPage.ts

```diff
@@ -0,0 +1,133 @@
+import { useRouter } from 'next/router'
+import { GetServerSideProps, Redirect } from 'next/types'
+
+export type ResolvedPathObject = {
+  remoteContainer: string | RemoteData,
+  modulePath: string,
+  /**
+   * In case of custom path resolver for parametrized routes,
+   * this is the key to the remote module.
+   * Example: path = '/storage/123', resolvedPath = '/storage/*'
+   */
+  resolvedPath: string
+}
+
+export type PathResolver = (path: string) => ResolvedPathObject
+
+function defaultPathResolver(path: string): ResolvedPathObject {
+  const trimmedPath = path.replace(/^\/+/, '')
+  const slashIndex = trimmedPath.indexOf('/')
+  const remoteContainer = slashIndex > -1 ? trimmedPath.substring(0, slashIndex) : trimmedPath
+  const modulePath = slashIndex > -1 ? trimmedPath.substring(slashIndex) : '/index'
+
+  return {
+    remoteContainer,
+    modulePath: `.${modulePath}`,
+    resolvedPath: path,
+  }
+}
+
+export type ErrorHandler = (error: unknown) => { redirect: Redirect } | { notFound: true }
+
+function defaultErrorHandler(error: unknown): { redirect: Redirect } | { notFound: true } {
+  console.error(error)
+  return {
+    notFound: true,
+  }
+}
+
+/**
+ * This function creates a dynamic page that is loaded from a remote defined by the path.
+ * @typedef {Object} CreateDynamicFederatedPageOptions
+ * @property pathResolver - function to obtain remote container name and module path from the page path, by default it's a simple path splitting
+ * @property errorHandler - function to handle errors, by default it just logs the error and returns 404
+ * @property suspenseFallback - fallback to render while remote module is loading
+ * TODO: remove injectScriptReplacement when it's fixed in @module-federation/utilities
+ * @property injectScriptReplacement - replacement for @module-federation/utilities/injectScript, which import is broken in @module-federation/nextjs-mf v6.4.0
+ */
+interface CreateDynamicFederatedPageOptions {
+  pathResolver?: PathResolver,
+  errorHandler?: ErrorHandler,
+  suspenseFallback?: React.ReactNode
+  injectScriptReplacement?: typeof injectScript,
+}
+
+export function createDynamicFederatedPage({
+  pathResolver = defaultPathResolver,
+  errorHandler = defaultErrorHandler,
+  suspenseFallback = null,
+  injectScriptReplacement = injectScript,
+}: CreateDynamicFederatedPageOptions = {}) {
+  const getRemoteModule = async (path: string) => {
+    const { remoteContainer, modulePath } = pathResolver(path)
+    const remoteContainerGlobal = typeof remoteContainer === 'string' ? remoteContainer : remoteContainer.global
+
+    // getModule() doesn't work on server-side
+    // and doesn't work on first render on client-side
+    if (typeof window === 'undefined' || !(window as any)[remoteContainerGlobal]) {
+      const container = await injectScriptReplacement(remoteContainer)
+      // @ts-ignore
+      return await container.get(modulePath)?.then((factory) => factory())
```

Comment:
Consider looking at my importDelegatedModule function from utilities. Something like that might be a little higher level. I also have things like getContainer and getModule that might offer some DRY